src/robusta/core/triggers/helm_releases_triggers.py

```diff
@@ -0,0 +1,196 @@
+    duration: Optional[int]
+    firing_release: Optional[HelmRelease]
+
+    def get_trigger_event(self):
+        return HelmReleasesTriggerEvent.__name__
+
+    def should_fire(self, event: TriggerEvent, playbook_id: str):
+        self.firing_release = None
+
+        if not isinstance(event, HelmReleasesTriggerEvent):
+            return False
+
+        if self.statuses and event.helm_release.info.status not in self.statuses:
+            return False
+        if self.names and event.helm_release.name not in self.names:
+            return False
+        if self.namespace and event.helm_release.namespace != self.namespace:
+            return False
+
+        last_deployed_utc = event.helm_release.info.get_last_deployed().astimezone(pytz.utc)
+
+        if self.duration:
+            # if the last deployed time is lesser than the `duration` time then dont append to the filtered list
+            now_utc = datetime.now().astimezone(pytz.utc)
+            time_delta = now_utc - last_deployed_utc
+            delta_seconds = time_delta.total_seconds()
+            if delta_seconds < self.duration:
+                return False
+
+        start_time_utc = ServerStart.get().astimezone(pytz.utc)
+        pod_start_time_delta_seconds = (start_time_utc - last_deployed_utc).total_seconds()
+        dont_fire = False
+
+        if event.helm_release.info.status in UNHEALTHY_STATUSES:
+            rate_limiter_id = f"{event.helm_release.namespace}:{event.helm_release.name}"
+        else:
+            rate_limiter_id = f"{event.helm_release.namespace}:{event.helm_release.name}-{last_deployed_utc.isoformat()}"
+            # if the server start time is greater than the last deployement time of the release then dont fire the trigger
+            # eg:   start -> 5pm, last_deployed -> 6pm, delta -> - 1hr. => fire
+            #       start -> 3pm, last_deployed -> 2pm, delta -> 1hr. => dont_fire
+            dont_fire = pod_start_time_delta_seconds > 0
+
+        # Perform a rate limit for this release according to the rate_limit parameter
+        if dont_fire:
+            return False
+
+        can_fire = RateLimiter.mark_and_test(
+            f"{self.trigger_name}_{playbook_id}",
+            rate_limiter_id,
+            self.rate_limit,
+        )
+
+        if can_fire:
+            logging.info(f"triggering a helm release event: {self.trigger_name}")
+
+        self.firing_release = event.helm_release
+
+        return can_fire
+
+    def build_execution_event(
+            self, event: HelmReleasesTriggerEvent, sink_findings: Dict[str, List[Finding]]
+    ) -> Optional[ExecutionBaseEvent]:
+        if not isinstance(event, HelmReleasesTriggerEvent):
+            return
+
+        return HelmReleasesEvent(
+            helm_release=self.firing_release,
+        )
+
+    @staticmethod
+    def get_execution_event_type() -> type:
+        return HelmReleasesEvent
+
+
+class OnHelmReleaseUnhealthyTrigger(HelmReleaseBaseTrigger):
```

Comment:
no need for the `On`
By the trigger naming convention it should be: `HelmReleaseUnhealthyTrigger`